temporal-sdk/src/main/java/io/temporal/common/converter/ProtobufJsonPayloadConverter.java

```diff
@@ -59,11 +59,15 @@ public String getEncodingType() {
     }
 
     try {
+      String messageTypeName = ((MessageOrBuilder) value).getDescriptorForType().getFullName();
       String data = printer.print((MessageOrBuilder) value);
       return Optional.of(
           Payload.newBuilder()
               .putMetadata(
                   EncodingKeys.METADATA_ENCODING_KEY, EncodingKeys.METADATA_ENCODING_PROTOBUF_JSON)
+              .putMetadata(
```

Comment:
Should we also require the message type to be accurate in `fromData` if it's present?

Reply:
Pro: that could help catch bugs, if the wrong message type is accidentally used
Con: less flexibility, if users intend to use a different message type

I'm inclined toward flexibility

Reply:
The problem is wrongness. Assert the correctness of metadata, be it message type, encoding, or whatever seems very reasonable. At the least, maybe a warning could be issued.